cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsReporterSampler.java

```diff
@@ -54,6 +55,9 @@
   private Consumer<String, CruiseControlMetric> _metricConsumer;
   private String _metricReporterTopic;
   private Set<TopicPartition> _currentPartitionAssignment;
+  // Due to delay introduced by KafkaProducer and network, the metric record's event time may be smaller than append
+  // time at broker side, sampler should take this delay into consideration when aggregating metric records int samples.
```

Comment:
Is it possible to be more clear about what happens if this constraint is violated? -- i.e. the record is discarded if its timestamp plus this additional acceptable delay is smaller than the start time of the sampling period?